src/test/utils/overridesHelper.ts

```diff
@@ -1,61 +1,101 @@
 import { Weather, WeatherType } from "#app/data/weather";
 import { Biome } from "#app/enums/biome";
 import * as Overrides from "#app/overrides";
-import { vi } from "vitest";
+import { MockInstance, vi } from "vitest";
+import GameManager from "#test/utils/gameManager";
+import { MysteryEncounterType } from "#enums/mystery-encounter-type";
+import * as overrides from "#app/overrides";
 
 /**
  * Helper to handle overrides in tests
  */
 export class OverridesHelper {
-  constructor() {}
+  game: GameManager;
+  constructor(game: GameManager) {
+    this.game = game;
+  }
 
   /**
    * Override the encounter chance for a mystery encounter.
    * @param percentage the encounter chance in %
+   * @returns spy instance
    */
-  mysteryEncounterChance(percentage: number) {
+  mysteryEncounterChance(percentage: number): MockInstance {
     const maxRate: number = 256; // 100%
     const rate = maxRate * (percentage / 100);
-    vi.spyOn(Overrides, "MYSTERY_ENCOUNTER_RATE_OVERRIDE", "get").mockReturnValue(rate);
+    const spy = vi.spyOn(Overrides, "MYSTERY_ENCOUNTER_RATE_OVERRIDE", "get").mockReturnValue(rate);
     this.log(`Mystery encounter chance set to ${percentage}% (=${rate})!`);
+    return spy;
+  }
+
+  /**
+   * Override the encounter that spawns for the scene
+   * @param encounterType
+   * @returns spy instance
+   */
+  mysteryEncounter(encounterType: MysteryEncounterType): MockInstance {
```

Comment:
```suggestion
  mysteryEncounter(encounterType: MysteryEncounterType) {
```
let it self-inject the type. That is a more specific version of MockInstance